### PR TITLE
:boom: New gql api for apps and overall code refactoring

### LINKED ISSUE #132: saagieapi apps API is not compatible with newer versions of Saagie platform
When using `saagie_api.apps.list_for_project_minimal` and `saagie_api.apps.list_for_project` on recently updated Saagie platforms, errors with the following message: "labWebApps operation is no longer supported. Use project instead."

### PR DESCRIPTION
**Pull request by TFellet**
- Modified all apps gql queries to be compatible with the new version of Saagie product
- Modified all apps methods to correspond to new gql queries
- Added a new method to validate alerting specifically for apps
- Removed all deprecated methods directly inside saagie_api.py
- Removed deprecated method for creating non-graph pipelines
- Improved pylint rating (changing variables names, removing nested conditions, ...)
- Renamed saagie client inside integration test file from saagie to saagie_api to be coherent with the rest of the package
- Updated graphql schema for unit testing
- fix #132

### CODE REVIEW

**Comment by saagie-service** [comment]
When this PR will be merged, we'll need to update this doc to specify which Saagie version is compatible with this release https://github.com/saagie/api-saagie#compatibility-with-your-saagie-platform

File: saagieapi/pipelines/pipelines.py [hunk 1]
```diff
@@ -161,35 +160,6 @@ def get_instance(self, pipeline_instance_id: str, pprint_result: Optional[bool]
             pprint_result=pprint_result,
         )
 
-    @deprecation.deprecated(
```
Comment (tquiviger) [inline saagieapi/pipelines/pipelines.py:164]: Do you plan to do a major release that brings breaking changes? If not, users will be affected by the removal of these deprecated methods and won't be warned...

File: saagieapi/saagie_api.py [hunk 2]
```diff
@@ -381,488 +380,3 @@ def get_technology_name_by_id(self, technology_id: str) -> (str, str):
             return "", ""
         repo_name, tech_name = technology_label_with_repo_name[0]
         return repo_name, tech_name
-
```
Comment (tquiviger) [inline saagieapi/saagie_api.py:384]: Do you plan to do a major release that brings breaking changes? If not, users will be affected by the removal of these deprecated methods and won't be warned...

Comment (TFellet) [reply saagieapi/saagie_api.py:384]: Major release planned

File: saagieapi/pipelines/pipelines.py (hunk 1 above)
Comment (TFellet) [reply saagieapi/pipelines/pipelines.py:164]: Major release planned

File: saagieapi/apps/apps.py [hunk 3]
```diff
@@ -434,69 +376,135 @@ def run(self, app_id: str) -> Dict:
...
-                        all(elem in LIST_EXPOSED_PORT_FIELD for elem in ep.keys()) for ep in exposed_ports
-                    )
-                    if check_every_key:
-                        return True
-                    return False
-                return False
-            return False
         return True
 
+    @staticmethod
```
[the remark this answers is not in the document]
Comment (TFellet) [reply saagieapi/apps/apps.py:429]: They have differents status list, the code around the function has been changed (now all parameters are optionals when editing an alerting, you can change only the status list or the email), and it has a new parameter, logins.
With some modifications in jobs.py we could use this function for both, and the valid status list would be a parameter.

**Review by guillaumeprevost76 — APPROVED** [review]

**Review by tquiviger — APPROVED** [review]

**Pull request merged**